analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/InvalidCastToInterface.CSharp8.cs

```diff
@@ -16,14 +16,14 @@ public void NullCoalescenceAssignmentResult_Null()
     {
         int? i = null;
         i ??= null;
-        var r1 = (int)i; // FIXME Non-compliant
+        var r1 = (int)i; // Noncompliant
     }
 
     public void NullCoalescenceAssignmentResult_NotNull()
     {
         int? i = null;
         i ??= 1;
-        var r1 = (int)i;
+        var r1 = (int)i;    // Noncompliant FP FIXME, this doesn't call property int?.HasValue, but method int?.HasValue.get
```

Comment:
Issue created #7308

Reply:
The root cause is in the engine.
`i` is `null`, so `??= 1` will assign `i= 1` that is not null for sure.
Therefore `(int)i` cast will succeed and rule should not raise => FP

Reply:
"It's an FP" 🤦 
My brain went bananas.